scripts/powershell/copy-secrets.ps1

```diff
@@ -0,0 +1,133 @@
+
+  .PARAMETER SourceVaultName
+  Specifies the name of the source key vault. Mandatory.
+
+  .PARAMETER TargetVaultName
+  Specifies the name of the target key vault. Mandatory.
+
+  .PARAMETER Force
+  Forces the script to copy regardless if secret exist in target Key vault. Can be omitted.
+
+  .EXAMPLE
+  This example shows how to copy all secrets from source to target vault within the same subscription.
+  If secret exists in target Key vault, it will not be copied.
+  .\Copy-AzKeyVaultSecret.ps1 -SourceSubscriptionId <String> -SourceVaultName <String> -TargetVaultName <String>
+
+  .EXAMPLE
+  Similar to example above, this shows how to copy all secrets from source to target vault within the same subscription.
+  Secret will be copied even if it already exist in target Key vault.
+  .\Copy-AzKeyVaultSecret.ps1 -SourceSubscriptionId <String> -SourceVaultName <String> -TargetVaultName <String> -Force
+
+  .EXAMPLE
+  This example shows how to copy all secrets when vaults reside in different Azure Subscriptions:
+  .\Copy-AzKeyVaultSecret.ps1 -SourceSubscriptionId <String> -TargetSubscriptionId <String> -SourceVaultName <String> -TargetVaultName <String>
+#>
+
+param (
+  [Parameter(Mandatory = $true)]
+  [string]$SourceSubscriptionId,
+
+  [Parameter(Mandatory = $false)]
+  [string]$TargetSubscriptionId,
+
+  [Parameter(Mandatory = $true)]
+  [string]$SourceVaultName,
+
+  [Parameter(Mandatory = $true)]
+  [string]$TargetVaultName,
+
+  # Force overwrite existing secrets
+  [Parameter(Mandatory = $false)]
+  [switch]$Force
+)
+
+$IpAddress = (Invoke-RestMethod -Uri "https://api.ipify.org")
+$IpAddressRange = "$IpAddress/32"
+Write-Information "Current IP address: $IpAddress"
+
+$Context = Set-AzContext -Subscription $SourceSubscriptionId
+Write-Information "Current subscription: $($Context.Subscription.Name)"
+
+$SourceVault = Get-AzKeyVault -VaultName $SourceVaultName
+$AddNetworkRule = $SourceVault.NetworkAcls.IpAddressRanges -notcontains $IpAddressRange
+
+try {
+  # Add IP adderss to source Key vault
+  if ($AddNetworkRule) {
+    $null = Add-AzKeyVaultNetworkRule -VaultName $SourceVaultName -IpAddressRange $IpAddressRange
+  }
+
+  # Get list of secrets names
+  $SourceVaultSecretNames = (Get-AzKeyVaultSecret -VaultName $SourceVaultName).Name
+
+  $SourceVaultSecrets = @()
+  $SourceVaultSecretNames | ForEach-Object {
+    $SourceVaultSecrets += (Get-AzKeyVaultSecret -VaultName $SourceVaultName -Name $_)
+  }
+}
+catch {
+  Write-Error "An error occurred: $_"
+}
+finally {
+  # Remove IP address from source Key vault
+  if ($AddNetworkRule) {
+    $null = Remove-AzKeyVaultNetworkRule -VaultName $SourceVaultName -IpAddressRange $IpAddressRange
+  }
+}
+
+# Switch context to target subscription if $TargetSubscriptionId has value
+if (![string]::IsNullOrEmpty($TargetSubscriptionId)) {
+  $Context = Set-AzContext -Subscription $TargetSubscriptionId
+  Write-Information "Target subscription: $($Context.Subscription.Name)"
+}
```

Comment:
I think we can clarify the logic here a bit by comparing source and target subscription IDs:

```suggestion
# If a separate target subscription has been specified, switch context
if ($TargetSubscriptionId -ne $SourceSubscriptionId) {
  $Context = Set-AzContext -Subscription $TargetSubscriptionId
  Write-Information "Target subscription: $($Context.Subscription.Name)"
}
```